flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala

```diff
@@ -205,6 +206,26 @@ object ScalarFunctions {
     new String(data)
   }
 
+
+  /**
+    * Returns a string resulting from replacing all substrings
+    * that match the regular expression with replacement.
+    */
+  def regexp_replace(str: String, regex: String, replacement: String): String = {
```

Comment:
Why not using `str.replaceAll()` here? Besides, I think the `replacement` string should be preprocessed with `Matcher.quoteReplacement()`.

Reply:
OK, will use this method.

Reply:
@xccui Thank you for your suggestion, I replaced it with String#replace, my original code and the current code work fine, but I think we can't call `Matcher.quoteReplacement()` on the regular expression string, it will cause Some test cases are not normal, such as `r$`, we will not get `fooba`

Reply:
Hi @yanghua , what I mean is the `replacement` string. You should try something like `"foobar".regexpReplace("oo", "$")` and `"foobar".regexpReplace("oo", "\\")`.